src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp

```diff
@@ -97,213 +92,409 @@ Position const SpawnBat[6] =
+        {
+            // charge the nearest player that is at least 8 yards away (charge min distance)
+            if (Unit* target = SelectTarget(SelectTargetMethod::MinDistance, 0, -8.0f, false, false))
+            {
+                LOG_DEBUG("scripts.ai", "boss_jeklik::UpdateAI:: Charge successful (target: {})", target->GetName());
+                DoCast(target, SPELL_CHARGE);
+                AttackStart(target);
+            }
+            else
+            {
+                LOG_DEBUG("scripts.ai", "boss_jeklik::UpdateAI:: Charge failed (no target available)");
+            }
+            context.Repeat(15s, 30s);
+        // Pierce Armor
+        }).Schedule(5s, 15s, PHASE_ONE, [this](TaskContext context)
+        {
+            DoCastVictim(SPELL_PIERCE_ARMOR);
+            context.Repeat(20s, 30s);
+        // Blood Leech
+        }).Schedule(5s, 15s, PHASE_ONE, [this](TaskContext context)
+        {
+            DoCastVictim(SPELL_BLOOD_LEECH);
+            context.Repeat(10s, 20s);
+        // Sonic Burst
+        }).Schedule(5s, 15s, PHASE_ONE, [this](TaskContext context)
+        {
+            DoCastVictim(SPELL_SONIC_BURST);
+            context.Repeat(20s, 30s);
+        // Swoop
+        }).Schedule(20s, PHASE_ONE, [this](TaskContext context)
         {
+            DoCastVictim(SPELL_SWOOP);
+            context.Repeat(20s, 30s);
+        // Spawn Cave Bats
+        }).Schedule(30s, PHASE_ONE, [this](TaskContext context)
+        {
+            Talk(EMOTE_SUMMON_BATS);
+            if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
+            {
+                for (uint8 i = 0; i < 6; ++i)
+                {
+                    if (Creature* bat = me->SummonCreature(NPC_BLOODSEEKER_BAT, SpawnBat[i], TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
+                    {
+                        bat->AI()->AttackStart(target);
+                    }
+                }
+            }
+            context.Repeat(30s);
+        });
+
+        //
+        // Phase 2 (@ 50% health)
+        //
+        ScheduleHealthCheckEvent(50, [&]
+        {
+            LOG_DEBUG("scripts.ai", "boss_jeklik:: PHASE TWO");
             me->RemoveAurasDueToSpell(SPELL_BAT_FORM);
             DoResetThreatList();
-            events.SetPhase(PHASE_TWO);
-            events.CancelEventGroup(PHASE_ONE);
 
-            events.ScheduleEvent(EVENT_CURSE_OF_BLOOD, 5s, 15s, PHASE_TWO);
-            events.ScheduleEvent(EVENT_SHADOW_WORD_PAIN, 10s, 15s, PHASE_TWO);
-            events.ScheduleEvent(EVENT_PSYCHIC_SCREAM, 25s, 35s, PHASE_TWO);
-            events.ScheduleEvent(EVENT_MIND_FLAY, 10s, 30s, PHASE_TWO);
-            events.ScheduleEvent(EVENT_GREATER_HEAL, 25s, PHASE_TWO);
-            events.ScheduleEvent(EVENT_SPAWN_FLYING_BATS, 10s, PHASE_TWO);
+            scheduler.CancelGroup(PHASE_ONE);
 
-            return;
-        }
+            // Curse of Blood
```

Comment:
Not sure on precedent, but these comments feel redundant given the enumeration is being utilized.

Reply:
I guess this is my replacement for the clean view of the EventMap where I could easily see all the abilities together. At least this way all of them are in green text and I can quickly identify them. I find it visually difficult to track down the one enum value (when multiple are used) to get to the spell I care about.

If it's a show-stopper, I can remove them. IMO ACore could use some more comments for newbs like me trying to learn how this thing works, but I understand that's not the culture here and I'm trying to find the right balance. See: very few logs in this AI.

Let me know what you'd like me to do.

Reply:
This is what I'm talking about with EventMap - super easy to see the timing settings, and I can go right to the spell I care about. TaskScheduler makes that difficult and at least for me, the comments create structure that's not inherent in the TaskScheduler way.

![image](https://github.com/azerothcore/azerothcore-wotlk/assets/19712078/c7d75dca-f536-4d54-9a58-f64b7696aca0)

Reply:
Yeah agree with Bench here. Imo they shouldn't be needed when the spells already use the enum

Reply:
I understand why namespaces got objected to since it was a new coding practice, but I must admit that I don't understand the objection here. As a newb reading an AI for the first time, it would have made a huge difference to me if someone commented and explain WTF they were doing instead of just assuming I understood all their code.

Maybe I'm just not cut out to redo these if all the things I feel like improve the code aren't acceptable. Very frustrated.

I will remove them if that's what's required to get this accepted.

Reply:
Quite frankly, I feel it's just a personal preference thing and definitely don't think it should be a barrier to merge. I mean, it is just a comment after all, we've all seen worse comments in this codebase. At the end of the day, I certainly don't mind 'em being there.

Reply:
I pushed a commit that removes any comments that might be considered obvious. Let me know if that is closer to the level of commenting I should be aiming for.